### PR TITLE
Use new mail package instead of an unmintained one

### LINKED ISSUE #18664: gomail library unmaintained
The gomail library used by Gitea is out of date. Transfer from me from the channel ([Matrix](https://matrix.to/#/%23gitea%3Amatrix.org/%24Kh1hQlkz2u0spp6ihaF_SCgScOBdJ-iPtbx-FwhHUac?via=matrix.org&via=libera.chat&via=maclemon.at))

> Gitea seems to use https://github.com/go-gomail/gomail which is unmaintained since 2016, and the "actively maintained" fork appears unmaintained, too: https://github.com/go-mail/mail/
> Do you consider this an issue for you? I looked at some Gitea issues, and some cases like different timeouts / retries is also a feature request to both frameworks, but apparently never implemented. Does a switch make sense, or is the current state perfectly fine?

---

Discussion already started here:

> FYI
> 
> * https://github.com/go-gomail/gomail/issues/163
> * https://github.com/go-gomail/gomail/issues/150
> 
> Maybe we should maintain gomail by ourselves, or switch to another one. Then we may not need hacky code.
> 
> 
> Some active forks:
> 
> * https://github.com/Shopify/gomail (Shopify is a big company)
> * https://github.com/DeedleFake/gomail (new)

_Originally posted by @wxiaoguang in https://github.com/go-gitea/gitea/issues/18075#issuecomment-1000053400_

### PR DESCRIPTION
**Pull request by lunny**
Resolve #18664

### CODE REVIEW

**Comment by Neustradamus** [comment]
@lunny: Excellent, good job!
If all devs were like you! :)

**Comment by Neustradamus** [comment]
@lunny: Excellent, good job if all devs were like you! :)
Hope that the team will review quickly and release a new build asap ^^

Linked to:
- https://github.com/go-gitea/gitea/issues/18664
- first PR: https://github.com/go-gitea/gitea/pull/32618
- Second PR: https://github.com/go-gitea/gitea/pull/32682

**Review by lafriks — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Comment by Neustradamus** [comment]
@lunny: Good job, thanks!

File: services/mailer/sender/sender.go [hunk 1]
```diff
@@ -19,9 +21,18 @@ func send(sender Sender, msgs ...*Message) error {
...
-		goMsgs = append(goMsgs, msg.ToMessage())
+		m := msg.ToMessage()
+		froms := m.GetFrom()
+		to, err := m.GetRecipients()
+		if err != nil {
+			return err
+		}
+
+		// TODO: implement sending from multiple addresses
+		if err := sender.Send(froms[0].Address, to, m); err != nil {
```
Comment (wxiaoguang) [inline services/mailer/sender/sender.go:33]: It causes panics like "Panic during startup #35768"

I don't know why the `froms` can be empty. Maybe you can figure out the root cause and/or improve " Fix various bugs #36139" / add more comments

**Pull request merged**